### PR TITLE
Supabase integration and misc features

### PR DESCRIPTION
**Pull request by nerfZael**
- Saving all data for the API in the supabase db
- Support for multiple apps with their own API keys
- Authentication for apps
- Support for multiple agent accounts per user (apps can have many users)
- Tracking for submitted batches of transactions

Code that tracks dev mode and switching between cli and API versions is a bit wonky and could be refactored in the future

### CODE REVIEW

File: supabase/config.toml [hunk 1]
```diff
@@ -0,0 +1,178 @@
+# A string used to distinguish different Supabase projects on the same host. Defaults to the
```
Comment (dOrgJelli) [inline supabase/config.toml:1]: Can we add instructions in the README for how to start / stop the DB for local development?

**Review by dOrgJelli — APPROVED** [review]
:clap: This looks fantastic

File: supabase/config.toml (hunk 1 above)
Comment (nerfZael) [reply supabase/config.toml:1]: Added API.md here: https://github.com/polywrap/AutoTx/pull/254/files#diff-b161966f7254af9c9c5382e859495c69622843e4602a6bb4208e08134e660121
And liked to it from the main README.md

**Comment by nerfZael** [comment]
/workflows/benchmarks agents/token/send 3

**Pull request merged**